src/components/search/SearchService.js

```diff
@@ -56,74 +113,68 @@ goog.require('ga_reframe_service');
+        // Parse Degrees Minutes Seconds
+        var matchDMS = query.match(regexpDMS);
+        // 0.7 is a magic number that defines if a majority of characters
+        // are detected.
+        // If not, try another rule (e.g. DM or DD)
+        if (matchDMS && matchDMS.length === 2 &&
+           (query.length * 0.7) <= (matchDMS[0].length + matchDMS[1].length)) {
+          left = parseDMS(matchDMS[0]);
+          right = parseDMS(matchDMS[1]);
+          if (right && left) {
+            coord = sortCoordinates(left, right);
+            position = ol.proj.transform(coord,
+                'EPSG:4326', gaGlobalOptions.defaultEpsg);
+            if (ol.extent.containsCoordinate(extent, position)) {
+              return $q.when(roundCoordinates(position));
+            }
+          }
+        }
+
+        // Parse Degrees EPSG:4326 notation
+        var matchD = query.match(regexpD);
+        if (matchD && matchD.length === 3) {
+          // we only care about coordinates in Switzerland
+          left = parseFloat(matchD[1]);
+          right = parseFloat(matchD[2]);
+          coord = sortCoordinates(left, right);
+          position = ol.proj.transform(coord, 'EPSG:4326',
+              gaGlobalOptions.defaultEpsg);
           if (ol.extent.containsCoordinate(extent, position)) {
             return $q.when(roundCoordinates(position));
           }
         }
 
-        var match = query.match(regexpCoordinate);
-        if (match) {
-          var left = parseFloat(match[1].replace(/'/g, ''));
-          var right = parseFloat(match[2].replace(/'/g, ''));
-          // Old school entries like '600 000 200 000'
-          if (match[3] != null) {
-            left = parseFloat(match[1].replace(/'/g, '') +
-                              match[2].replace(/'/g, ''));
-            right = parseFloat(match[4].replace(/'/g, '') +
-                               match[5].replace(/'/g, ''));
+        // Parse Degrees Minutes
+        var matchDM = query.match(regexpDM);
+        if (matchDM && matchDM.length === 5) {
+          left = parseInt(matchDM[1]) + parseFloat(matchDM[2]) / 60.0;
+          right = parseInt(matchDM[3]) + parseFloat(matchDM[4]) / 60.0;
+          coord = sortCoordinates(left, right);
+          position = ol.proj.transform(coord, 'EPSG:4326',
+              gaGlobalOptions.defaultEpsg);
+          if (ol.extent.containsCoordinate(extent, position)) {
+            return $q.when(roundCoordinates(position));
           }
-          position = [left > right ? left : right,
-            right < left ? right : left];
+        }
+
+        // Parse swiss coordinates
+        var match = query.match(regexpCoordinate);
+        // Matches new school entries like '2 600 000 1 200 000'
+        // and old school entries like '600 000 200 000'
+        if (match && match.length === 4) {
+          left = sanitizeCoordinate(match[1]);
+          right = sanitizeCoordinate(match[3]);
+          position = [
```

Comment:
May use `sortCoordinates `

Reply:
fixed

Reply:
in fact you can't it reorder in A bad way:

in 4326 northing>easting
in swiss easting>NORTHING

I've removed the comment